library/src/main/java/bio/terra/stairway/Control.java

```diff
@@ -0,0 +1,583 @@
+        NamedParameterPreparedStatement statement =
+            new NamedParameterPreparedStatement(connection, sb.toString())) {
+      DbUtils.startReadOnlyTransaction(connection);
+      if (status != null) {
+        statement.setString("status", status);
+      }
+      List<Control.Flight> flightList = flightQuery(statement, offset, limit);
+      DbUtils.commitTransaction(connection);
+      return flightList;
+    }
+  }
+
+  public List<Control.Flight> listOwned(int offset, int limit) throws SQLException {
+    StringBuilder sb = new StringBuilder();
+    sb.append("SELECT").append(flightSelectFrom);
+    sb.append(" WHERE stairway_id IS NOT NULL ");
+    sb.append(flightOrderPage);
+
+    try (Connection connection = dataSource.getConnection();
+        NamedParameterPreparedStatement statement =
+            new NamedParameterPreparedStatement(connection, sb.toString())) {
+      DbUtils.startReadOnlyTransaction(connection);
+      List<Control.Flight> flightList = flightQuery(statement, offset, limit);
+      DbUtils.commitTransaction(connection);
+      return flightList;
+    }
+  }
+
+  public Control.Flight getFlight(String flightId) throws SQLException {
+    StringBuilder sb = new StringBuilder();
+    sb.append("SELECT").append(flightSelectFrom);
+    sb.append(" WHERE flightid = :flightid ");
+    sb.append(flightOrderPage);
+
+    try (Connection connection = dataSource.getConnection();
+        NamedParameterPreparedStatement statement =
+            new NamedParameterPreparedStatement(connection, sb.toString())) {
+      statement.setString("flightid", flightId);
+      DbUtils.startReadOnlyTransaction(connection);
+      List<Control.Flight> flightStateList = flightQuery(statement, 0, 1);
+      DbUtils.commitTransaction(connection);
+      if (flightStateList.isEmpty()) {
+        throw new IllegalArgumentException("Unknown flight id " + flightId);
+      }
+      return flightStateList.get(0);
+    }
+  }
+
+  public Control.Flight flightDisown(String flightId)
+      throws SQLException, DatabaseOperationException, InterruptedException, FlightException {
+    Control.Flight flight = getFlight(flightId);
+    if (flight.getStairwayId().isEmpty()) {
+      throw new IllegalStateException("Flight is not owned: " + flightId);
+    }
+
+    // Dummy up a flight context and use that to call the flight doa to disown.
+    // When we have fixed serdes, we can use real flight contexts
+    FlightMap fakeFlightMap = new FlightMap();
+    FlightContext fakeFlightContext = new FlightContext(fakeFlightMap, null, null);
+    fakeFlightContext.setFlightStatus(FlightStatus.READY);
+    fakeFlightContext.setFlightId(flightId);
+    flightDao.exit(fakeFlightContext);
+
+    return getFlight(flightId);
+  }
+
+  public Control.Flight forceFatal(String flightId) throws SQLException {
+    // We do not assume anything about the current state. We just force to a FATAL
+    // disowned state so no attempt will be made to recover.
```

Comment:
re: the `TODO`, do you mean so we can tell the difference between a forced FATAL and a natural FATAL?

Reply:
Yes. I am not sure what the right thing is, but it seems like we should be keeping an audit trail somewhere of the changes made. Especially in a production situation. I made a ticket for this and referenced it here in the TODO.